candle-metal-kernels/src/indexing.metal

```diff
@@ -26,6 +44,7 @@ METAL_FUNC void index(
     // No need to check for zero we're only allowing unsized. 
     */ 
     const size_t src_i = left_rank_i * src_dim_size * right_size + input_i * right_size + right_rank_i; 
+    const size_t strided_src_i = contiguous ? src_i : get_strided_index(src_i, src_dim_size, src_dims, src_strides);
```

Comment:
Shouldn't `strided_src_i` be used instead of `src_i` or am I missing something? And if it's an actual issue, shouldn't a test have a failed on this this?

Reply:
That's a great point... My test is not doing the right thing with simulating a strided buffer perhaps. Let me take a look at some of our other strided function tests 🤔

Reply:
Okay latest push with the test has a case that actually fails as it should with the first commit on this branch, do you mind taking a peak at the test case and let me know if that's the right direction for this?